hack/generator/pkg/astmodel/std_references.go

```diff
@@ -54,18 +54,18 @@ var (
 	KubernetesResourceType         = MakeTypeName(GenRuntimeReference, "KubernetesResource")
 	ConvertibleSpecInterfaceType   = MakeTypeName(GenRuntimeReference, "ConvertibleSpec")
 	ConvertibleStatusInterfaceType = MakeTypeName(GenRuntimeReference, "ConvertibleStatus")
-	ResourceReferenceTypeName      = MakeTypeName(GenRuntimeReference, "ResourceReference")
-	KnownResourceReferenceTypeName = MakeTypeName(GenRuntimeReference, "KnownResourceReference")
+	ResourceReferenceType          = MakeTypeName(GenRuntimeReference, "ResourceReference")
```

Comment:
I wrote a medium-length comment here about how this was confusing because the variable name suggested it was a type but it was a pointer to a type (the structure of the type was somewhere else). Then I realised that the type of this variable is `TypeName` so users can tell it's the name and not the thing itself. 
Now I think maybe you could drop the `-Type` suffix at the end as well? Although I guess the suffix here is the kind of thing the name refers to?

Reply:
I considered dropping the `Type` but decided not to, because the variable doesn't contain (for example) a `KnownResourceReference` (it's not a reference to any one resource, known or otherwise), it instead contains a reference to the `Type` of all of those references.